Lombiq.OrchardCoreApiClient/ApiClient.cs

```diff
@@ -4,62 +4,32 @@
-    private DateTime _expirationDateUtc = DateTime.MinValue;
-    private Token _tokenResponse;
+    protected readonly ApiClientSettings _apiClientSettings;
+    private Lazy<IOrchardCoreApi> LazyOrchardCoreApi => new(() => RestClient.For<IOrchardCoreApi>(_httpClient));
+    private ConfigurableCertificateValidatingHttpClientHandler _certificateValidatingHandler;
+    protected HttpClient _httpClient;
+
     public IOrchardCoreApi OrchardCoreApi => LazyOrchardCoreApi.Value;
 
-    public ApiClient(ApiClientSettings apiClientSettings) => _apiClientSettings = apiClientSettings;
+    public ApiClient(ApiClientSettings apiClientSettings)
+    {
+        _apiClientSettings = apiClientSettings;
 
-    private IOrchardCoreApi GetOrchardCoreApi(Uri defaultTenantUri) =>
-        RestClient.For<IOrchardCoreApi>(defaultTenantUri, async (request, _) =>
+        _certificateValidatingHandler = new ConfigurableCertificateValidatingHttpClientHandler(_apiClientSettings);
+#pragma warning disable CA5399 // HttpClient is created without enabling CheckCertificateRevocationList
+        _httpClient = new HttpClient(_certificateValidatingHandler)
         {
-            if (request.Headers.Authorization == null)
-            {
-                throw new ApiClientException("The request header should contain an authorization token.");
-            }
-
-            if (_expirationDateUtc < DateTime.UtcNow.AddSeconds(60))
-            {
-                var tokenResponse = await RestClient
-                    .For<IOrchardCoreAuthorizationApi>(defaultTenantUri)
-                    .TokenAsync(
-                        new Dictionary<string, string>
-                        {
-                            ["grant_type"] = "client_credentials",
-                            ["client_id"] = _apiClientSettings.ClientId,
-                            ["client_secret"] = _apiClientSettings.ClientSecret,
-                        });
-
-                tokenResponse.ResponseMessage.EnsureSuccessStatusCode();
-
-                if (tokenResponse.GetContent().Error != null)
-                {
-                    throw new ApiClientException(
-                        $"API client setup failed. An error occurred while retrieving an access token: " +
-                        tokenResponse.GetContent().Error);
-                }
-
-                int tokenExpiration = int.Parse(tokenResponse.GetContent().ExpiresIn, CultureInfo.InvariantCulture);
-
-                _expirationDateUtc = DateTime.UtcNow.AddSeconds(tokenExpiration);
-
-                _tokenResponse = tokenResponse.GetContent();
-            }
-
-            request.Headers.Authorization = new AuthenticationHeaderValue(
-                request.Headers.Authorization.Scheme, _tokenResponse.AccessToken);
-        });
+            BaseAddress = _apiClientSettings.DefaultTenantUri,
+        };
+#pragma warning restore CA5399
```

Comment:
https://github.com/Lombiq/.NET-Analyzers/blob/dev/Lombiq.Analyzers/Docs/UsingAnalyzersDuringDevelopment.md#:~:text=Always%20add%20a%20justification.

Also in `ConfigurableCertificateValidatingHttpClientHandler`.